File: user-springboot-admin/admin/src/main/java/com/artsail/iot/model/domain/IotDeviceTypeCommand.java
package com.artsail.iot.model.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;

@Data
@TableName("iot_device_type_command")
public class IotDeviceTypeCommand implements Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long deviceTypeId;
    private String commandKey;
    private String commandName;
    private String confirmText;
    private Integer isStop;
    private Integer sortOrder;
    private Integer isActive;
}

File: user-center-admin/src/services/api/iot-device-type-command.ts
/**
 * 设备类型指令管理 API
 * 对应 iot_device_type_command 表
 * 统一入口：/iot/device-type-config/commands
 */
import { request } from '@umijs/max';
import type { BaseResponse } from '@/types/common';

export interface IoTDeviceTypeCommand {
  id?: number;
  deviceTypeId: number;
  commandKey: string;
  commandName: string;
  confirmText?: string;
  isStop?: number;
  sortOrder?: number;
  isActive?: number;
}

const BASE = '/iot/device-type-config/commands';

/** 获取指令列表 */
export async function getCommandList(deviceTypeId?: number) {
  return request<BaseResponse<IoTDeviceTypeCommand[]>>(BASE, {
    method: 'GET',
    params: { deviceTypeId },
  });
}

/** 新增指令 */
export async function createCommand(body: Partial<IoTDeviceTypeCommand>) {
  return request<BaseResponse<boolean>>(BASE, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    data: body,
  });
}

/** 更新指令 */
export async function updateCommand(id: number, body: Partial<IoTDeviceTypeCommand>) {
  return request<BaseResponse<boolean>>(BASE + '/' + id, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    data: body,
  });
}

/** 删除指令 */
export async function deleteCommand(id: number) {
  return request<BaseResponse<boolean>>(BASE + '/' + id, {
    method: 'DELETE',
  });
}
